second draft commit to ebasseri fork (#15101)
microsoftgraph/microsoft-graph-docs: api-reference/beta/api/identityprotectionroot-list-serviceprincipalriskdetections.md

---
title: "List servicePrincipalRiskDetections"

> description: "Get the servicePrincipalRiskDetection resources from the servicePrincipalRiskDetections navigation property."

Reviewer: ```suggestion
description: "Retrieve the properties of a collection of servicePrincipalRiskDetection objects."
```

Reviewer: @ebasseri, bumping this change suggestion.

author: "ebasseri"
ms.localizationpriority: medium
ms.prod: "identity-and-sign-in"
doc_type: apiPageType
---

# List servicePrincipalRiskDetections
Namespace: microsoft.graph

[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]

> Retrieve the properties of a collection of servicePrincipalRiskDetection objects.

Reviewer: ```suggestion
Retrieve the properties of a collection of [servicePrincipalRiskDetection](../resources/serviceprincipalriskdetection.md) objects.
```

Reviewer: @ebasseri, bumping this change suggestion.

>**Note:** You must have an Azure AD Premium P1 or P2 license to use the servicePrincipalRiskDetection API.

## Permissions
One of the following permissions is required to call this API. To learn more, including how to choose permissions, see [Permissions](/graph/permissions-reference).

|Permission type|Permissions (from least to most privileged)|
|:---|:---|
|Delegated (work or school account)|IdentityRiskyServicePrincipal.Read.All, IdentityRiskyServicePrincipal.ReadWrite.All|
|Delegated (personal Microsoft account)|Not supported.|
|Application|IdentityRiskyServicePrincipal.Read.All, IdentityRiskyServicePrincipal.ReadWrite.All|

## HTTP request

<!-- {
  "blockType": "ignored"
}
-->
``` http
GET /identityProtection/servicePrincipalRiskDetections
```

## Optional query parameters

> This method supports `$filter` and `$select` to customize the query response. See the example later in this topic.

Reviewer: ```suggestion
This method supports the `$filter` and `$select` OData query parameters to help customize the response. For general information, see [OData query parameters](/graph/query-parameters).
```

Reviewer: @ebasseri, bumping this change suggestion.

  "@odata.type": "Collection(microsoft.graph.servicePrincipalRiskDetection)"
}
-->
<!-- {
  "blockType": "response",
  "truncated": true,
  "@odata.type": "microsoft.graph.servicePrincipalRiskDetection"
}
-->
``` http
HTTP/1.1 200 OK
Content-Type: application/json

{
  "value": {
    "@odata.type": "#microsoft.graph.servicePrincipalRiskDetection",
    "id": "2856d6e87c5c3a74021ff70291fa68107570c150d8dc145bdea5",
    "requestId": "null",
    "correlationId": "null",
    "riskEventType": "investigationsThreatIntelligence",
    "riskState": "atRisk",
    "riskLevel": "high",
    "riskDetail": "none",
    "source": "IdentityProtection",
    "detectionTimingType": "offline",
    "activity": "servicePrincipal",
    "tokenIssuerType": "AzureAD",
    "ipAddress": "null",
    "location": "null",
    "activityDateTime": "2021-10-26T00:00:00Z",
    "detectedDateTime": "2021-10-26T00:00:00Z",
    "lastUpdatedDateTime": "2021-10-26T16:28:17.8202975Z)",
    "servicePrincipalId": "99b8d28b-11ae-4e84-9bef-0e767e286grg",
    "servicePrincipalDisplayName": "Contoso App",
    "appId": "0grb38ac-a572-491d-a9db-b07197643457",
    "keyIds": [
      "9d9fea30-d8e3-481b-b57c-0ef569a989e5"
    ],

>     "additionalInfo": "[{\"Key\":\"alertUrl\",\"Value\":null}]"

Reviewer: GLOBAL: I don't think null values are supposed to be returned in quotes. Kindly confirm and resolve in this and other examples in this PR.

Reviewer: @ebasseri, bumping this change suggestion.

  }
}
```

### Example 2: Get a specific risk detection object

#### Request
<!-- {
  "blockType": "request",
  "name": "get_serviceprincipalriskdetection"
}
-->
``` http
GET https://graph.microsoft.com/beta/identityProtection/servicePrincipalRiskDetections/{servicePrincipalRiskDetectionId}
```

#### Response
>**Note:** The response should include a single object.
<!-- {
  "blockType": "response",
  "truncated": true,
  "@odata.type": "microsoft.graph.servicePrincipalRiskDetection"
}
-->
``` http
HTTP/1.1 200 OK
Content-Type: application/json

> 

Reviewer: Same comment about quoting null values.

Reviewer: @ebasseri, bumping this change suggestion.

    "servicePrincipalDisplayName": "Contoso App",
    "appId": "0grb38ac-a572-491d-a9db-b07197643457",
    "keyIds": [
      "9d9fea30-d8e3-481b-b57c-0ef569a989e5"
    ],
    "additionalInfo": "[{\"Key\":\"alertUrl\",\"Value\":null}]"
  }
}
```

### Example 3: List risk detections and filter the results

#### Request
The following example shows how to use `$filter` to get the collection of service principal risk detections where the risk level is medium or the risk event type is investigationsThreatIntelligence.

<!-- {
  "blockType": "request",
  "name": "list_filter_serviceprincipalriskdetection"
}
-->
``` http
GET https://graph.microsoft.com/beta/identityProtection/servicePrincipalRiskDetections?$filter=riskEventType eq 'investigationsThreatIntelligence' or riskLevel eq 'medium'
```

#### Response
Here is an example of the response.
<!-- {
  "blockType": "response",
  "truncated": true,
  "@odata.type": "microsoft.graph.servicePrincipalRiskDetection"
}
-->
``` http
HTTP/1.1 200 OK
Content-Type: application/json

{
  "value": {
    "@odata.type": "#microsoft.graph.servicePrincipalRiskDetection",
    "id": "2856d6e87c5c3a74021ff70291fa68107570c150d8dc145bdea5",

>     "requestId": "null",

Reviewer: Same comment about quoting null values.

Reviewer: @ebasseri, bumping this change suggestion.
